backend/schema/questions.sql

```diff
@@ -0,0 +1,10 @@
+CREATE TYPE difficulties AS ENUM ('easy', 'medium', 'hard');
+
+CREATE TABLE questions (
+    questionId SERIAL PRIMARY KEY,
+    questionTitle VARCHAR(256) NOT NULL UNIQUE,
+    categories VARCHAR(256)[] NOT NULL,
+    difficulty difficulties NOT NULL,
+    attempts INT DEFAULT 0,
+    createdAt TIMESTAMP DEFAULT CURRENT_TIMESTAMP
+);
```

Comment:
were there any reasons you decided to split question details into 2 separate databases? 

would it be better to have all details in 1 store to reduce complexity?

Reply:
should we move attempts to a questionAttempts table to minimise updates?

Reply:
This attempt is to store the number of people who have attempted a question so I think it should be here since the counter is only linked to the one question ID in the same row.

Reply:
If we are counting attempts by unique users, is there a way to know when to increase the count?

Reply:
So when a new row is added to the user history table, this counter here can be incremented. Either:
- The api manually increments it
- An SQL trigger will update it automatically

Reply:
hmm, in that case should we just use a filter and count to derive attempts from the userHistory table?

Reply:
We can do that as well, I just created that column in case we need it. We can remove it next time if we don't require it.